test/stdlib/SwiftObjectNSObject.swift

```diff
@@ -39,8 +39,24 @@ class D : C {
   @objc override class func cClassOverride() -> Int { return 8 }
 }
 
+class E : Equatable {
```

Comment:
Would be good to have two subclasses, and test all combinations of: two instances of same subclass, two instances of different subclasses, instance of subclass and instance of superclass.

Reply:
I've significantly extended the test cases.  This mostly involved adding some helpers that checked the Swift behavior then delegated to an Obj-C test function that verified the behavior matches Swift.